module/platform/OpenCR/HardwareIO/src/hardwareio/create_watchdogs.cpp

```diff
@@ -0,0 +1,106 @@
+            startup();
+            return;
+        });
+    }
+
+    NUClear::threading::ReactionHandle HardwareIO::create_packet_watchdog() {
+        return on<Watchdog<PacketWatchdog, 20, std::chrono::milliseconds>, Sync<PacketWatchdog>>().then([this] {
+            // This is a hacky fix because the watchdog is not disabled quickly enough at the beginning.
+            // This may be related to the out of order packets with Sync within NUClear. This should be
+            // fixed in a later version of NUClear.
+            if (model_watchdog.enabled()) {
+                log<NUClear::WARN>(
+                    "Packet watchdog cannot be enabled while model watchdog is enabled. You may see this "
+                    "warning at the start of the program. This is expected as the watchdog reaction may "
+                    "still be disabling.");
+                packet_watchdog.disable();
+                return;
+            }
+
+            // Check what the hangup was
+
+            int num_packets_dropped        = 0;                 // keep track of how many we dropped
+            NUgus::ID first_dropped_packet = NUgus::ID::NO_ID;  // keep track in case we have a chain
+
+            // The result of the assignment is 0 (NUgus::ID::NO_ID) if we aren't waiting on
+            // any packets, otherwise is the nonzero ID of the timed out device
+            for (NUgus::ID dropout_id; (dropout_id = queue_item_waiting()) != NUgus::ID::NO_ID;) {
+
+                // Delete the packet we're waiting on
+                packet_queue[dropout_id].erase(packet_queue[dropout_id].begin());
+
+                // notify with ID and servo name
+                log<NUClear::WARN>(
+                    fmt::format("Dropped packet from ID {} ({})", int(dropout_id), nugus.device_name(dropout_id)));
+
+                // if this is the first packet, set our flag
+                if (num_packets_dropped == 0) {
+                    first_dropped_packet = dropout_id;
+                }
+
+                // increment our counter
+                num_packets_dropped++;
+            }
+
+            // if this is the first packet then send a warning
+            if (num_packets_dropped > 1) {
+                log<NUClear::WARN>(
+                    fmt::format("NOTE: A dropped response packet by a dynamixel device in a SYNC READ/WRITE "
+                                "chain will cause all later packets (of higher ID) to be dropped. Consider "
+                                "checking cables for ID {} ({})",
+                                int(first_dropped_packet),
+                                nugus.device_name(first_dropped_packet)));
+            }
+
+            // Send a request for all servo packets, only if there were packets dropped
+            // In case the system stops for some other reason, we don't want the watchdog
+            // to make it automatically restart
+            if (num_packets_dropped > 0) {
```

Comment:
Just a naïve question that won't stop approval, but would `num_packets_dropped` ever be zero here? I know that it _can_, but what I am really driving at is what scenario would make that happen. I am just thinking aloud. It may be worth at least thinking about this hypothetical scenario in the wider context of these elusive OpenCR bugs.

Reply:
I'd be more worried about possibly flushing parts of or whole valid dynamixel packets that would lead to more dropped packets